### ISSUE TITLE
Airflow/Cognito Integration

### ISSUE THREAD

**Issue by LucaCinquini** [unity-sds/unity-project-management#212]
We need to have Users log in to Airflow using Cognito account.

**Comment by mike-gangl** [comment]
Gherkin should include

login via UI (manual)
token login via airflow API
token login via OGC endpoint

**Comment by rtapella** [comment]
login via unity-sds-client

**Comment by LucaCinquini** [comment]
System Validation Criteria:
o Unity developers that have a Cognito account can login to the Airflow UI via a web browser
o Successful execution of the Gherkin tests for:
- Cognito token login via airflow API
- Cognito token login via OGC endpoint

**Comment by GodwinShen** [comment]
@nikki-t successfully demo'd Cognito integration, work with @jpl-btlunsfo to work out the call back.

**Comment by LucaCinquini** [comment]
It was decided that the Cognito authentication will be handled by the top level shared services proxy.

What is working right now for SPS:
- The top-level shared service proxy is protected by Cognito authentication
- HTTP(s) requests are correctly proxied from top-level proxy to mid-level proxy to Airflow

What still needs to be implemented:
- We need to be able to use the Cognito token for Airflow and OGS API requests
- Once the above is working, remove the Airflow login from the mid-proxy and prevent direct access to the mid-proxy

**Comment by brianlee731** [comment]
@jpl-btlunsfo please provide a status/blcoker regarding this task. Thank you.

**Comment by brianlee731** [comment]
Need to update SPS Integration test before closing this task.

**Comment by jpl-btlunsfo** [comment]
SPS integration tests are updated and properly using Cognito auth- this issue can probably be closed...

**Comment by LucaCinquini** [comment]
Yes we can close this ticket. We closed the corresponding one in the SPS repo.
